qendpoint-core/src/main/java/com/the_qa_company/qendpoint/core/util/BitUtil.java

```diff
@@ -64,9 +64,10 @@ public static void writeLowerBitsByteAligned(long value, long numbits, OutputStr
 	public static int select1(long value, int rank) {
 		int bitpos = 0;
 		while (rank > 0 && value != 0) {
-			rank -= value & 1;
-			bitpos++;
-			value >>>= 1;
+			int trailingZeros = Long.numberOfTrailingZeros(value);
```

Comment:
This is faster. Not certain why, but I assume that the Long.numberOfTrailingZeros(value) method uses a SIMD instruction to check multiple bits in a single operation.